ml-prims/src/common/cuml_allocator.hpp

```diff
@@ -0,0 +1,35 @@
+/*
```

Comment:
Knitpicks!
1. can you remove the 'cuml_' prefix from the filename, please?
2. Everything inside ml-prims is currently under `MLCommon` namespace. So, in the interest of keeping everything uniform, can you also update the namespace, please?

Reply:
I am open for other suggestions but I have added the `cuml_` prefix because the file needs to be part of the public interface as it is included by `cuML.hpp` for a binary build of the library it therefore needs to end up in the include directory. Since `allocator.hpp` is probably a very common name for an include file we need to avoid ambiguity here. We can either do putting it in a subfolder of an include path or by prefixing the file name. For the former we would need a concept how to manage include paths during development and after deployment. Do we have that in place? 
Regarding the namespace I have kept `ML` as I found other occurrences of `ML` in `ml-prims`. Thus asking for confirmation before I change it to `MLCommon`.

Reply:
> re: namespace

Please go ahead with `MLCommon` itself. Later, I'll find those other occurrences and update them accordingly.

Reply:
> cuml prefix

ah... that's a tricky one!
1. keeping this name kind of suggests that there's a dependency from ml-prims to cuml.
2. changing its name means we'll have trouble exposing it to public interface.

Reply:
OK. I changed the namespace. Regarding the filename, include path issue. I think this is a topic on its own mostly orthogonal to this PR. Are you OK if I file a issue to track it and we defer that discussion to a later time?

Reply:
@dantegd do you have a better suggestion here for the filename?

If not, let's just keep the filename as it is and adjust it later as per the future needs.